drivers/led/Kconfig.gpio

```diff
@@ -3,6 +3,7 @@
 
 config LED_GPIO
 	bool "GPIO LED driver"
-	depends on GPIO && $(dt_compat_enabled,gpio-leds)
+	default y
+	depends on GPIO && DT_HAS_GPIO_LEDS_ENABLED
```

Comment:
This is a bit of a special case, I think, as `compatible = "gpio-leds" can either be accessed via the GPIO API or via the dedicated GPIO LEDs driver.

Reply:
Not sure I follow what you are suggestion we change here?

Reply:
I suggest we do not enable this particular driver based on the presence of a DTS compatible since most users will likely want to use the GPIO API instead of the dedicated LED driver.

Reply:
Do you think they will have CONFIG_LED=y set?

Reply:
Well, when you put it that way... I guess we can enable it by default based on the DT.